api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/PatientTranslatorImpl.java

```diff
@@ -134,7 +134,7 @@ public org.openmrs.Patient toOpenmrsType(@Nonnull org.openmrs.Patient currentPat
 		notNull(currentPatient, "The existing Openmrs Patient object should not be null");
 		notNull(patient, "The Patient object should not be null");
 		
-		currentPatient.setUuid(patient.getId());
+		currentPatient.setUuid(patient.getIdElement().getIdPart());
```

Comment:
This needs some more guards around it because this is used for both new and existing patients, i.e., the `Patient` object is not guaranteed to have an `idElement` nor for that to have an `idPart`. This is why I suggested an `if / else` guard on Talk.

Reply:
Donr